src/meta/src/controller/fragment.rs

```diff
@@ -60,13 +60,16 @@ use crate::stream::SplitAssignment;
 use crate::MetaResult;
 
 impl CatalogControllerInner {
-    /// List all fragment vnode mapping info
+    /// List all fragment vnode mapping info for all CREATED streaming jobs.
     pub async fn all_running_fragment_mappings(
         &self,
     ) -> MetaResult<impl Iterator<Item = FragmentParallelUnitMapping> + '_> {
         let fragment_mappings: Vec<(FragmentId, FragmentVnodeMapping)> = Fragment::find()
+            .join(JoinType::InnerJoin, fragment::Relation::Object.def())
+            .join(JoinType::InnerJoin, object::Relation::StreamingJob.def())
             .select_only()
             .columns([fragment::Column::FragmentId, fragment::Column::VnodeMapping])
+            .filter(streaming_job::Column::JobStatus.eq(JobStatus::Created))
```

Comment:
Only notify fragment mapping for `CREATED` streaming jobs, consistent with changes in create ddls.